python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -0,0 +1,191 @@
+#
+
+import numpy as np
+from scipy.stats import sigmaclip
+from scipy.fftpack import fft2, ifft2, ifftshift
+from scipy.ndimage.filters import convolve
+
+import lsst.afw.image as afwImage
+import lsst.meas.algorithms as measAlg
+import lsst.afw.math as afwMath
+
+__all__ = ("performExposureDecorrelation")
+
+
+def computeClippedImageStats(im, low=3, high=3):
+    """! Utility function for sigma-clipped array statistics.
+    @param im   A numpy array. Should work on any dimensions/shape.
+    @param low  Lower bound factor of sigma clipping. Default is 3.
+    @param high  Upper bound factor of sigma clipping. Default is 3.
+    @return sigma-clipped mean and std of input array
+    """
+    _, low, upp = sigmaclip(im, low=low, high=high)
+    tmp = im[(im > low) & (im < upp)]
+    mean1 = np.nanmean(tmp)
+    sig1 = np.nanstd(tmp)
+    return mean1, sig1
+
+
+def getImageGrid(im):
+    """! Convenience function for creating a grid of pixel coordinates covering a 2-d array,
+    using `numpy.meshgrid()`
+    @param im input 2-d numpy array
+    @return two numpy arraysm each with same shape as `im`, containing x- and y- coordinate
+    grid.
+    """
+    xim = np.arange(np.int(-np.floor(im.shape[0]/2.)), np.int(np.floor(im.shape[0]/2)))
+    yim = np.arange(np.int(-np.floor(im.shape[1]/2.)), np.int(np.floor(im.shape[1]/2)))
+    x0im, y0im = np.meshgrid(xim, yim)
+    return x0im, y0im
+
+
+def computeDecorrelationCorrectionKernel(kappa, sig1=0.2, sig2=0.2):
+    """! Compute the Lupton/ZOGY post-conv. kernel for decorrelating an
+    image difference, based on the PSF-matching kernel.
+    @param kappa  A matching kernel array derived from Alard & Lupton PSF matching
+    @param sig1   Average sqrt(variance) of template image used for PSF matching
+    @param sig2   Average sqrt(variance) of science image used for PSF matching
+    @return a 2-d numpy.array containing the correction kernel
+
+    @note As currently implemented, kappa is a static (single) kernel.
+    """
+    def post_conv_kernel_ft2(kernel, sig1=1., sig2=1.):
+        kft = fft2(kernel)
+        return np.sqrt((sig1**2 + sig2**2) / (sig1**2 + sig2**2 * kft**2))
+
+    def post_conv_kernel2(kernel, sig1=1., sig2=1.):
+        kft = post_conv_kernel_ft2(kernel, sig1, sig2)
+        out = ifft2(kft)
+        return out
+
+    pck = post_conv_kernel2(kappa, sig1=sig2, sig2=sig1)
+    pck = ifftshift(pck.real)
+
+    # I think we may need to "reverse" the PSF, as in the ZOGY (and Kaiser) papers...
+    # This is the same as taking the complex conjugate in Fourier space before FFT-ing back to real space.
+    if False:  # TBD: figure this out. For now, we are turning it off.
+        pck = pck[::-1, :]
+
+    return pck
+
+
+def computeCorrectedDiffimPsf(kappa, im2_psf, sig1=0.2, sig2=0.2):
+    """! Compute the (decorrelated) difference image's new PSF:
+    post_conv_psf = phi_1(k) * sym.sqrt((sig1**2 + sig2**2) / (sig1**2 + sig2**2 * kappa_ft(k)**2))
```

Comment:
I suggest moving this equation to the next line and providing a reference to a paper where it can be found. Also, what is "sym" in "sym.sqrt"?